### PR TITLE
FIX- 896/product type support 5

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
* convert bug
* force inline in convert
* zip traits changes
* read/write

### CODE REVIEW

File: include/eve/algo/concepts/eve_iterator.hpp [hunk 1]
```diff
@@ -55,6 +57,7 @@ namespace eve::algo
     std::totally_ordered_with<I, unaligned_t<I>> &&
     std::totally_ordered_with<I, partially_aligned_t<I>> &&
     requires(I i, std::ptrdiff_t n) {
+       { std::declval<typename I::value_type>() };
```
Comment (jfalcou) [inline include/eve/algo/concepts/eve_iterator.hpp:60]: you don't need the declval.
`typename I::value_type;` is enough.
See Type Requirements at https://en.cppreference.com/w/cpp/language/constraints

File: include/eve/algo/traits.hpp [hunk 2]
```diff
@@ -106,12 +106,13 @@ template <typename Traits, typename T>
     rbr::get_type_t<Traits, force_cardinal_key,
     eve::expected_cardinal_t<iteration_type_t<Traits, T>>>;
 
-  template <typename User, typename Default>
-  constexpr auto default_to(traits<User> const& user, traits<Default> const& defaults)
+
+  inline constexpr auto default_to =
```
Comment (jfalcou) [inline include/eve/algo/traits.hpp:110]: why a lambda instead of a function here ?

Comment (DenisYaroshevskiy) [reply include/eve/algo/traits.hpp:110]: To be able to pass into kumi::fold_right.

I was thinking to do a struct but seemed like lambda is enough.

I want to clean up all functions at some point

Comment (jfalcou) [reply include/eve/algo/traits.hpp:110]: OK

**Review by jfalcou — APPROVED** [review]

**Pull request merged**